portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -63,11 +81,55 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
     commentEntity.setProperty(COMMENT_TEXT, commentText);
     commentEntity.setProperty(TIME_STAMP, timeStamp);
 
+    // Used for testing, see TODO
+    System.out.println(getUploadedFileUrl(request, /* formInputElementName= */ "image"));
+
     DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
     datastore.put(commentEntity);
 
     response.sendRedirect(REDIRECT_URL);
   }
+
+  /** Returns a URL that points to the uploaded file, or null if the user didn't upload an image file. */
+  private String getUploadedFileUrl(HttpServletRequest request, String formInputElementName) {
+    BlobstoreService blobstoreService = BlobstoreServiceFactory.getBlobstoreService();
+    Map<String, List<BlobKey>> blobs = blobstoreService.getUploads(request);
+    List<BlobKey> blobKeys = blobs.get(formInputElementName);
+
+    // User submitted form without selecting a file, so we can't get a URL. (dev server)
```

Comment:
What does the comment (dev server) mean?

Reply:
dev server is the local server when testing. Changed from dev to local